components/gallery/GalleryForm.tsx

```diff
@@ -196,58 +199,55 @@ function GalleryForm(props: GalleryFormProps) {
     // setPinningServiceAccessToken("");
   }
 
-  const commitNewRoot = React.useCallback(
-    async (mediaGalleryItem: MediaGalleryItem) => {
-      if (!geoWebContent) {
-        return;
-      }
-      const assetId = new AssetId({
-        chainId: `eip155:${NETWORK_ID}`,
-        assetName: {
-          namespace: "erc721",
-          reference: licenseAddress.toLowerCase(),
-        },
-        tokenId: new BN(selectedParcelId.slice(2), "hex").toString(10),
-      });
-      const ownerId = new AccountId(
-        AccountId.parse(ceramic.did?.parent.split("did:pkh:")[1] ?? "")
-      );
-      const mediaGallery = await geoWebContent.raw.getPath("/mediaGallery", {
-        ownerId,
-        parcelId: assetId,
-      });
-      const rootCid = await geoWebContent.raw.resolveRoot({
-        ownerId,
-        parcelId: assetId,
-      });
-      const cidStr = mediaGalleryItem.content;
-      const mediaObject: MediaObject = {
-        name: mediaGalleryItem.name ?? "",
-        content: CID.parse(cidStr ?? ""),
-        encodingFormat: mediaGalleryItem.encodingFormat as Encoding,
-      };
-      const newRoot = await geoWebContent.raw.putPath(
-        rootCid,
-        selectedMediaGalleryItemIndex !== null
-          ? `/mediaGallery/${selectedMediaGalleryItemIndex}`
-          : `/mediaGallery/${mediaGallery.length}`,
-        mediaObject,
-        { parentSchema: "MediaGallery", pin: true }
-      );
-
-      await geoWebContent.raw.commit(newRoot, {
-        ownerId,
-        parcelId: assetId,
-      });
-      const newRootCid = await geoWebContent.raw.resolveRoot({
-        parcelId: assetId,
-        ownerId,
-      });
+  async function commitNewRoot(mediaGalleryItem: MediaGalleryItem) {
+    if (!geoWebContent) {
+      return;
+    }
+    const assetId = new AssetId({
+      chainId: `eip155:${NETWORK_ID}`,
+      assetName: {
+        namespace: "erc721",
+        reference: licenseAddress.toLowerCase(),
+      },
+      tokenId: new BN(selectedParcelId.slice(2), "hex").toString(10),
+    });
+    const ownerId = new AccountId(
+      AccountId.parse(ceramic.did?.parent.split("did:pkh:")[1] ?? "")
+    );
+    const mediaGallery = await geoWebContent.raw.getPath("/mediaGallery", {
+      ownerId,
+      parcelId: assetId,
+    });
+    const rootCid = await geoWebContent.raw.resolveRoot({
```

Comment:
There seem to be several calls to `resolveRoot` besides the one that is now in `useBasicProfile` as of #384 

Are each of these calls actually needed? Or could we consolidate them? The less calls we make to Ceramic the better at this point.

Reply:
Because `getPath` calls `resolveRoot` too we could just get the root here and pass it to `get` instead of `getPath`.
The same thing can now be done in #384, I acually was thinking of just doing it there but here we can save another call by passing the root CID returned by `putPath` to `setRootCid` at the bottom of the function.

So seems like we can consolidate and remove three unnecessary `resolveRoot` calls between here and `useBasicProfile`

Reply:
That sounds great. Let's use `get` where we can